pkg/handlers/handlers.go

```diff
@@ -285,14 +285,9 @@ func HandleFile(
 	}
 	defer func() {
 		// Ensure all data is read to prevent broken pipe.
-		_, copyErr := io.Copy(io.Discard, rdr)
```

Comment:
This is only necessary when the underlying reader is a pipe. If the reader isn't fully consumed, `cmd.Wait()` hangs indefinitely. Since `HandleFile` is used by other parts of the codebase with different readers, the logic to consume the reader (if it's a pipe) has been moved to the caller.